af_ktls.c

```diff
@@ -984,10 +988,14 @@ static inline ssize_t tls_peek_data(struct tls_sock *tsk, unsigned flags)
 			goto peek_failure;
 		}
 
+		if (peeked_size < KTLS_MIN_RECORD_SIZE(tsk)) {
+			ret = -EBADF;
```

Comment:
For EBADF that's fine, as long as we are only peeking KTLS_MIN_RECORD_SIZE

For the first case, EAGAIN is only for nonblocking sockets, as long as recvmsg does the right thing for blocking/nonblocking sockets and this error is only internal, we can pass around whatever we want.